packages/service-library/src/servicelib/async_utils.py

```diff
@@ -205,3 +212,40 @@ async def worker(in_q: Queue[QueueElement], out_q: Queue) -> None:
         return wrapper
 
     return decorator
+
+
+def with_delay(
+    delay: datetime.timedelta,
+) -> Callable[
+    [Callable[P, Coroutine[Any, Any, R]]], Callable[P, Coroutine[Any, Any, R]]
```

Comment:
TIP: perhaps it is more readable creating an alias for the types?

```python
CoroutineFunc: TypeAlias = Callable[P, Coroutine[Any, Any, R]]

def with_delay():
     def _decorator(coro: CoroutineFunc[P, R] ):
         @wraps(coro)
         async  def _wrapper(*args: P.args, **kwargs: P.kwargs) -> R:
                      ...
        return _wrapper
   return _decorator


```

Moreover. I am not sure you need to annotate the output of the decorator here. I thought this was automatically deduced by the static tooling... but ...  Am I missing something?

Reply:
it is not really automatically deduced. therefore we have many decorators that break the tooling..